packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol

```diff
@@ -172,8 +172,8 @@ contract OptimisticAssertor is Lockable, OptimisticAssertorInterface, Ownable {
             int256 resolvedPrice =
                 _getOracle(assertionId).getPrice(identifier, assertion.assertionTime, _stampAssertion(assertionId)); // Revert if price not resolved.
 
-            assertion.settlementResolution = resolvedPrice == 1e18;
-            address bondRecipient = assertion.settlementResolution ? assertion.proposer : assertion.disputer;
+            assertion.settlementResolution = assertion.useDisputeResolution ? resolvedPrice == 1e18 : false;
```

Comment:
Looks good. 
Then maybe [this line](https://github.com/UMAprotocol/protocol/blob/f0b58ae6bb1dfa521e11a06991cd6190685c548c/packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol#L129) can be removed as we already do the checks here?